src/libraries/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -475,34 +477,87 @@ internal BigInteger(int n, uint[]? rgu)
         /// </summary>
         /// <param name="value">The absolute value of the number</param>
         /// <param name="negative">The bool indicating the sign of the value.</param>
-        internal BigInteger(uint[] value, bool negative)
+        /// <param name="transfersOwnership">true if the BigInteger may assume ownership of the provided <paramref name="value"/>; otherwise, false.</param>
+        internal BigInteger(uint[] value, bool negative, bool transfersOwnership)
         {
-            if (value == null)
-                throw new ArgumentNullException(nameof(value));
+            Debug.Assert(value != null);
 
             int len;
 
-            // Try to conserve space as much as possible by checking for wasted leading uint[] entries
-            // sometimes the uint[] has leading zeros from bit manipulation operations & and ^
+            // Try to conserve space as much as possible by checking for wasted leading uint entries
+            // (sometimes the uint[] has leading zeros from bit manipulation operations & and ^).
             for (len = value.Length; len > 0 && value[len - 1] == 0; len--);
 
             if (len == 0)
+            {
                 this = s_bnZeroInt;
+            }
             // Values like (Int32.MaxValue+1) are stored as "0x80000000" and as such cannot be packed into _sign
             else if (len == 1 && value[0] < kuMaskHighBit)
             {
                 _sign = (negative ? -(int)value[0] : (int)value[0]);
                 _bits = null;
-                // Although Int32.MinValue fits in _sign, we represent this case differently for negate
-                if (_sign == int.MinValue)
+                if (_sign == int.MinValue) // although Int32.MinValue fits in _sign, we represent this case differently for negate
+                {
                     this = s_bnMinInt;
+                }
             }
             else
             {
                 _sign = negative ? -1 : +1;
-                _bits = new uint[len];
-                Array.Copy(value, _bits, len);
+                if (transfersOwnership && value.Length == len)
```

Comment:
Do we know how often this is false because `value.Length != len`?

I'm mostly interested for data towards my `BigInteger.Builder` proposal 😄

Reply:
Don't know.  Seems like it it's reasonably common as the result of certain operators?